### PR TITLE
completed ticket 3 testing. Set up app.js, controller, controller err…

### PR DESCRIPTION
**Pull request by TheCodedJourney**
…ors and also topic models. Listed controllers and models with their respective categories as have seen there are a few more categories to come which I will need to test and split out. Have split out for readability and organisation purposes. Ready for feedback

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const request = require('supertest')
+
+const app = require('../app')
+const db = require('../db/connection')
+const testData = require('../db/data/test-data/index')
+const seed = require('../db/seeds/seed') 
+
+afterAll(() => db.end)
```
Comment (roisinoreilly) [inline __tests__/api.test.js:8]: you need to invoke db.end() here

File: __tests__/api.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+beforeEach(() => seed(testData))
+
+describe('API testing', () => {
+    describe('GET Requests', () => {
+        test('200 - Should return correct values within the topics API. Format should be an array of objects for slug and description properties', () => {
+            return request(app)
+            .get('/api/topics')
+            .expect(200)
+            .then(({ body: { topics } }) => {
+                expect(topics).toBeInstanceOf(Array);
```
Comment (roisinoreilly) [inline __tests__/api.test.js:18]: is this line necessary if you're successfully iterating through topics a few lines later?

File: __tests__/api.test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                expect(topics).toHaveLength(3);
```
Comment (roisinoreilly) [inline __tests__/api.test.js:19]: good job checking the length of the array here

File: __tests__/api.test.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    });
+        test('404 - It should return an error when the wrong path is provided', () => {
+            return request(app)
+            .get('/api/topical')
+            .expect(404)
+            .then(({body : {msg}}) => {
+             expect(msg).toBe("404 Not Found")   
+            })
+        });
+        // test('GET /api/description', () => {
```
Comment (roisinoreilly) [inline __tests__/api.test.js:38]: make sure you remove all commented out code before pushing

File: app.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const express = require("express")
+
+const {getTopicPath} = require('./controllers/controllers.topic')
+const {error404} = require('./controllers/controllers.errors')
+
+const app = express()
+app.use(express.json());
```
Comment (roisinoreilly) [inline app.js:7]: can you remove this line up until the point where we need to use it to keep it strictly TDD

File: controllers/controllers.errors.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+const error404 = (request, response, next)=> {
+    response.status(404).send({msg: "404 Not Found"})
```
Comment (roisinoreilly) [inline controllers/controllers.errors.js:2]: nicely handled, have a think about including a 500 error handler too

File: controllers/controllers.topic.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+const {selectTopicPath} = require('../models/models.topics')
+
+
+const getTopicPath = (request, response, next) => {
+  selectTopicPath()
+      .then((topics) => {
+        response.status(200).send({ topics });
```
Comment (roisinoreilly) [inline controllers/controllers.topic.js:7]: nice and neat

**Review by roisinoreilly — APPROVED** [review]
Great job! There's a couple of small changes I'd like you to make but once they're done feel free to merge away :)

File: __tests__/api.test.js (hunk 2 above)
Comment (TheCodedJourney) [reply __tests__/api.test.js:18]: Good point, will remove this

File: __tests__/api.test.js (hunk 3 above)
Comment (TheCodedJourney) [reply __tests__/api.test.js:19]: Thank you!

File: __tests__/api.test.js (hunk 4 above)
Comment (TheCodedJourney) [reply __tests__/api.test.js:38]: Done ✅

File: app.js (hunk 5 above)
Comment (TheCodedJourney) [reply app.js:7]: just to confirm, you want to me to remove 4,5,6,7 ?

File: __tests__/api.test.js (hunk 1 above)
Comment (TheCodedJourney) [reply __tests__/api.test.js:8]: Done ✅

File: controllers/controllers.errors.js (hunk 6 above)
Comment (TheCodedJourney) [reply controllers/controllers.errors.js:2]: Done ✅

File: app.js (hunk 5 above)
Comment (roisinoreilly) [reply app.js:7]: just line 7 :)

Comment (TheCodedJourney) [reply app.js:7]: Done ✅

**Pull request merged**